src/engine/source/hlp2/src/parse_field.cpp

```diff
@@ -56,7 +56,7 @@ std::optional<Field> getField(std::string_view input,
                 }
                 else
                 {
-                    bool escaped = (last_escape_location + 1 == i);
```

Comment:
The doxygen documentation is missing, you can base it on the following to add it to the header file
```hpp
/**
 * @brief Parses a field from a string using the given delimiter, quote, and escape characters.
 * 
 * @param input The string to parse the field from.
 * @param delimiter The delimiter character used to separate fields.
 * @param quote The quote character used to enclose fields that contain delimiter characters.
 * @param escape The escape character used to escape quote or escape characters inside a field.
 * @param strict Whether strict parsing should be used. If true, fields not enclosed in quotes cannot contain quotes.
 * @return An optional Field object containing the parsed field, or std::nullopt if the input is invalid.
 */
```

Some additional comments might be useful to explain some design decisions or to clarify some implementation details.
Please analyze and consider the following code, What do you think?
```cpp
std::optional<Field> getField(std::string_view input,
                              const char delimiter,
                              const char quote,
                              const char escape,
                              bool strict)
{
    size_t last_escape_location = 0;

    if (input.empty())
    {
        // Empty field
        return Field {0, 0, false, false};
    }

    bool quote_opened {false};
    bool isEscaped = false;
    bool isQuoted = false;

    for (auto i = 0ul; i < input.size(); i++)
    {
        if (input[i] == delimiter && !quote_opened)
        {
            // Found delimiter outside quotes, return field
            return Field {0, i, isEscaped, isQuoted};
        }
        else
        {
            if (quote_opened && quote != escape && input[i] == escape)
            {
                // Handle escape character
                last_escape_location = i;
            }
            if (input[i] == quote)
            {
                if (!quote_opened)
                {
                    // Handle opening quote
                    quote_opened = true;
                    if (strict && (i > 1) && input[i - 1] != delimiter)
                    {
                        // Invalid field if strict parsing is enabled and there is no delimiter before opening quote
                        return {};
                    }
                    if (quote == escape)
                    {
                        last_escape_location = i;
                    }
                }
                else
                {
                    // Handle closing quote
                    bool escaped = (last_escape_location + 1 == i) && (i > 1);
                    isEscaped = isEscaped || escaped;
                    last_escape_location += (i - last_escape_location) * size_t(!escaped);
                    quote_opened = escaped || (input[i + 1] != delimiter);
                    isQuoted = true;
                }
            }
        }
    }
    // Unclosed quote
    if (quote_opened && input.back() != quote)
    {
        return {};
    }

    // Return field
    return Field {0, input.size(), isEscaped, isQuoted};
};

```

Reply:
I agree, I will replace it